webkitpy: Use WinCairoPort by default on Windows

fujii opened the pull request:
#### 4075f39b99867ed031472f80da844bc2cead2069
<pre>
webkitpy: Use WinCairoPort by default on Windows
<a href="https://bugs.webkit.org/show_bug.cgi?id=253912">https://bugs.webkit.org/show_bug.cgi?id=253912</a>

Reviewed by Jonathan Bedard.

With this change, we can invoke run-webkit-tests without --wincairo
switch on Windows. However, we can&apos;t remove --wincairo switch yet
because buildbot is still using it. I&apos;ll rename WinCairoPort to
WinPort in the future. However, there are some remaining tasks. I keep
the name WinCairoPort at the moment.

* Tools/Scripts/webkitpy/common/config/ports.py:
(DeprecatedPort.port):
(MacWK2Port):
(WinPort): Deleted.
(WinPort.run_webkit_tests_command): Deleted.
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(_set_up_derived_options):
* Tools/Scripts/webkitpy/port/factory.py:
(PortFactory):
(PortFactory._default_port):
* Tools/Scripts/webkitpy/port/factory_unittest.py:
(FactoryTest.test_win):

Canonical link: <a href="https://commits.webkit.org/261712@main">https://commits.webkit.org/261712@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f17710912f737db21a23363236183304c780a26f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112527 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21675 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1180 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4302 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/121088 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/116593 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/23013 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12836 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/5435 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118294 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/17111 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100295 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105582 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/115947 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/99043 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/840 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/46096 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/14017 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/877 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/95172 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14701 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/10260 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/20040 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52882 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8168 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16535 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (approved):
This change looks good.

Do we _want_ to remove the `--wincairo` switch? I think with have a `--mac` on MacOS, even though it's the default....

fujii committed:
webkitpy: Use WinCairoPort by default on Windows
https://bugs.webkit.org/show_bug.cgi?id=253912

Reviewed by Jonathan Bedard.

With this change, we can invoke run-webkit-tests without --wincairo
switch on Windows. However, we can't remove --wincairo switch yet
because buildbot is still using it. I'll rename WinCairoPort to
WinPort in the future. However, there are some remaining tasks. I keep
the name WinCairoPort at the moment.

* Tools/Scripts/webkitpy/common/config/ports.py:
(DeprecatedPort.port):
(MacWK2Port):
(WinPort): Deleted.
(WinPort.run_webkit_tests_command): Deleted.
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(_set_up_derived_options):
* Tools/Scripts/webkitpy/port/factory.py:
(PortFactory):
(PortFactory._default_port):
* Tools/Scripts/webkitpy/port/factory_unittest.py:
(FactoryTest.test_win):

Canonical link: https://commits.webkit.org/261712@main
